### DISCUSSION TITLE
Release 5.1.1

### DISCUSSION THREAD

**Discussion by Athanasius** [EDCD/EDMarketConnector#1163] [Announcement]
* We now test against, and package with, Python 3.9.5.

  **As a consequence of this we no longer support Windows 7.
  This is due to [Python 3.9.x itself not supporting Windows 7](https://www.python.org/downloads/windows/).
  The application (both EDMarketConnector.exe and EDMC.exe) will crash on startup due to a missing DLL.**

  This should have no other impact on users or plugin developers, other than the latter now being free to use features that were introduced since the Python 3.7 series.

  Developers can check the contents of the `.python-version` file in the source (it's not distributed with the Windows installer) for the currently used version in a given branch.

Release 5.1.1
===

The big change in this is adjustments to be in line with Journal changes in Elite Dangerous Odyssey 4.0.0.400, released 2021-06-10, with respect to the Odyssey materials Inventory.

**This update is mandatory if you want EDMarketConnector to update Inara.cz with your Odyssey inventory.**

* `ShipLockerMaterials` is dead, long live `ShipLocker`.  Along with other changes to how backpack inventory is handled we should now actually be able to fully track all Odyssey on-foot materials and consumables without errors.

* Inara plugin adjusted to send the new `ShipLocker` inventory to Inara.cz.  This is *still* only your *ship* inventory of Odyssey materials, not anything currently in your backpack whilst on foot.
  See [this issue](https://github.com/EDCD/EDMarketConnector/issues/1162) for some quotes from Artie (Inara.cz developer) about *not* including backpack contents in the Inara inventory.

* Errors related to sending data to EDDN are now more specific to aid in diagnoising issues.

* Quietened some log output if we encounter connection errors trying to utilise the Frontier CAPI service.

Translations
---
We believe that nothing should be worse in this version compared to 5.1.1, although a small tweak or two might have leaked through.

We'll be fully addressing translations in a near-future release after we've conclude the necessary code level work for the new system.  Nothing should change for those of you helping on OneSky, other than at most the 'comments' on each translation.  They should be more useful!

Pending that work we've specifically chosen *not* to update any translations in this release, so they'll be the same as released in 5.1.0.

Bug Fixes
---

* Handle where the `Backpack.json` file for a `Backpack` event is a zero length
  file.  Closes #1138.

* Fixed case of 'Selection' in 'Override Beta/Normal Selection' text on Settings > Configuration.  This allows translations to work.

Plugin Developers
---
* We've updated [Contributing.md](./Contributing.md) including:

  1. Re-ordered the sections to be in a more logcial and helpful order.
  1. Added a section about choosing an appropriate log level for messages.
  1. fstrings now mandatory, other than some use of `.format()` with respect to translated strings.

* [docs/Translations.md](./docs/Translations.md) updated about a forthcoming change to how we can programmatically check that all translation strings have a proper comment in 'L10n/en.template' to aid translators.

* `state` passed to `journal_entry()` now has `ShipLockerJSON` which contains the `json.load()`-ed data from the new 'ShipLocker.json' file.  We do attempt to always load from this file, even when the `ShipLocker` Journal event itself contains all of the data (which it does on startup, embark and disembark), so it *should* always be populated when plugins see any event related to Odyssey inventory.

<hr /><em>This discussion was created from the release <a href='https://github.com/EDCD/EDMarketConnector/releases/tag/Release/5.1.1'>Release 5.1.1</a>.</em>

**Comment by AndroidDataEvans** [comment]
i run ED on my win 7 pc, not going to change it yet at all, ive got a laptop which is win 10, networked, please provide a way, without usin programming tech talk, so i can send data to ED, when i am flying, it cant be that hard to bridge them surely, make a script or somthing

**Reply by Athanasius** [reply]
This is not a tech support service.

What you want to learn how to do is share the files from the Windows 7 PC to the Windows 10 one.  Then run EDMC on the Windows 10 one and change the Journal File Location in the settings to where they're shared into that machine.

**Comment by FusionJohn** [comment]
I had this installed V5.1.1 and running earlier today and now it won't connect. I tried to deauthorize and also uninstall / reinstall but still no connection. I don't get the FD page opening to authorize when I open EDMC. There is no connected info displayed.

**Reply by Athanasius** [reply]
> I had this installed V5.1.1 and running earlier today and now it won't connect. I tried to deauthorize and also uninstall / reinstall but still no connection. I don't get the FD page opening to authorize when I open EDMC. There is no connected info displayed.

Please open a [Bug Report](https://github.com/EDCD/EDMarketConnector/issues/new?assignees=&labels=bug%2C+unconfirmed&template=bug_report.md&title=) with all the required information, **including the requested log files**.  It's difficult, if not impossible, to diagnose such issues without the right information.

**Reply by FusionJohn** [reply]
OK, fair point. Thanks, I'll check it out.
